File: csharp/AdventOfCode/AdventOfCode2017/Day1.cs
namespace AdventOfCode.AdventOfCode2017
{
    public class Day1
    {
        public static int First(string input)
        {
            int acc = 0;
            for (int i = 0; i < input.Length; i++)
            {
                if (input[i] == input[(i + 1) % input.Length])
                    acc += int.Parse(input[i].ToString());
            }
            return acc;
        }

        public static int Second(string input)
        {
            int acc = 0;
            int halfLength = input.Length / 2;
            for (int i = 0; i < input.Length; i++)
            {
                if (input[i] == input[(i + halfLength) % input.Length])
                    acc += int.Parse(input[i].ToString());
            }
            return acc;
        }
    }
}

File: typescript/2017/day1.ts
import assert = require("assert");

function First(input: string) {
    let acc = 0;
    for (let i = 0; i < input.length; i++) {
        if(input[i] == input[(i + 1) % input.length])
            acc += Number(input[i]);
    }
    return acc;
}

assert(First("1122") === 3);
assert(First("1111") === 4);
assert(First("1234") === 0);
assert(First("91212129") === 9);
console.log("First tests passed");

function Second(input: string) {
    let acc = 0;
    for (let i = 0; i < input.length; i++) {
        if(input[i] == input[(i + input.length / 2) % input.length])
            acc += Number(input[i]);
    }
    return acc;
}

assert(Second("1212") === 6);
assert(Second("1221") === 0);
assert(Second("123425") === 4);
assert(Second("123123") === 12);
assert(Second("12131415") === 4);
console.log("Second tests passed");